components/public-api/gitpod/v1/prebuild.proto

```diff
@@ -0,0 +1,107 @@
+syntax = "proto3";
+
+package gitpod.v1;
+
+import "google/protobuf/timestamp.proto";
+import "gitpod/v1/pagination.proto";
+
+option go_package = "github.com/gitpod-io/gitpod/components/public-api/go/v1";
+
+service PrebuildService {
+    rpc StartPrebuild(StartPrebuildRequest) returns (StartPrebuildResponse) {}
+    rpc StopPrebuild(StopPrebuildRequest) returns (StopPrebuildResponse) {}
+
+    rpc GetPrebuild(GetPrebuildRequest) returns (GetPrebuildResponse) {}
+    rpc ListPrebuilds(ListPrebuildsRequest) returns (ListPrebuildsResponse) {}
+    rpc WatchPrebuild(WatchPrebuildRequest) returns (stream WatchPrebuildResponse) {}
+}
+
+message GetPrebuildRequest {
+    string prebuild_id = 1;
+    string workspace_id = 2;
+    string configuration_id = 3;
+    string branch = 4;
+}
+message GetPrebuildResponse {
+    Prebuild prebuild = 1;
+}
+
+message ListPrebuildsRequest {
+    PaginationRequest pagination = 1;
+    string prebuild_id = 2;
+    string configuration_id = 3;
+    string branch = 4;
+}
+
+message ListPrebuildsResponse {
+    PaginationResponse pagination = 1;
+    repeated Prebuild prebuilds = 2;
+}
+
+message StartPrebuildRequest {
+    string configuration_id = 1;
+    string branch = 2;
+}
+message StartPrebuildResponse {
+    string prebuild_id = 1;
+}
+
+message StopPrebuildRequest {
+    string prebuild_id = 1;
+    // waht for?
+    string configuration_id = 2;
+}
+message StopPrebuildResponse {
+}
+
+message WatchPrebuildRequest {
+    string prebuild_id = 1;
+    string workspace_id = 2;
+    string configuration_id = 3;
+}
+message WatchPrebuildResponse {
+    Prebuild prebuild = 1;
+}
+
+message Author {
+    string name = 1;
+    string avatar_url = 2;
+}
+
+message Commit {
+    string message = 1;
+    Author author = 2;
+    google.protobuf.Timestamp author_date = 3;
+    string sha = 4;
+}
+
+message Prebuild {
+    string id = 1;
+    string based_on_prebuild_id = 2;
+    string build_workspace_id = 3;
+
+    string configuration_id = 4;
+    string branch = 5;
+    Commit commit = 6;
+    string url = 7;
+
+    google.protobuf.Timestamp start_time = 8;
+    Author started_by = 9;
+
+    enum Status {
+        STATUS_UNSPECIFIED = 0;
+        STATUS_QUEUED = 1;
+        STATUS_BUILDING = 2;
+        STATUS_ABORTED = 3;
+        STATUS_TIMEOUT = 4;
+        STATUS_AVAILABLE = 5;
+        STATUS_FAILED = 6;
+    }
+    Status status = 10;
+
+    message ErrorDetails {
+        string message = 1;
+    }
+    // inspect in case of status FAILED, TIMEOUT or ABORTED
+    ErrorDetails details = 11;
```

Comment:
We could rather just call it message for now? And introduce details or conditions later when we need more fine grain details, i.e. failure codes.